lib/bootstrap_form/form_builder.rb

```diff
@@ -370,13 +382,19 @@ def form_group_builder(method, options, html_options = nil)
     end
 
     def convert_form_tag_options(method, options = {})
-      options[:name] ||= method
-      options[:id] ||= method
+      unless @options[:skip_default_ids]
+        options[:name] ||= method
+        options[:id] ||= method
+      end
       options
     end
 
     def generate_label(id, name, options, custom_label_col, group_layout)
-      options[:for] = id if acts_like_form_tag
+      # id is the caller's options[:id] at the only place this method is called.
+      # The options argument is a small subset of the options that might have
+      # been passed to generate_label's caller, and definitely doesn't include
+      # :id.
+      options[:for] = id if acts_like_form_tag || id
```

Comment:
I am also having trouble figuring out this part. If `id` is present, then it will be assigned regardless of whether `acts_like_form_tag` is true or false. Why is the `acts_like_form_tag` check needed at all, then?

Reply:
This was to address part of the consequence of no default IDs in Rails 5.1 `form_with`. I'll take this out, and let it be handled by #357. And yes, the `acts_like_form_tag` becomes redundant when I added `|| id`. "Doh!" I don't know what I was thinking. :-)